update claim doc (#372)
hats-finance/hats-contracts: docs/claims.md

## Details and restrictions

There are 3 "periods" relevant for the handling of a new claim:
1. Challenge Period: the first `challengePeriod` seconds after the submission of the claim 
1. Challenge Timeout Period: the `challengeTimeOutPeriod` seconds after the challenge period ended
1. Expiration: after `challengePeriod + challengeTimeoutPeriod` seconds after the submission of the claim

1. **SUBMISSION** 
  `submitClaim(beneficiary, bountyPercentage)` will create a new `claimId`.
   - `submitClaim` can only be called by the committee
   - `submitClaim` can only be called during a safety period
   - `submitClaim` can only be called if no other active claim exists
   - emergencyPause must not be in effect
1. **CHALLENGE** `challengeClaim(_claimId)` 
   - can only be called during the challengePeriod
   - can only be called during the challenge period
   - only the `arbitrator` or the `registry.owner` can challenge a claim (the registry.owner functions here as a "challenger of last resort")
   - can only be called if the claim has not been challenged yet

>    - `_claimId` must be the id of the currently active claim

Reviewer: too obvious?

Author: read it with a stresss on "currently active"

>    - during the challenge period and challenge timeout period:

Reviewer: ```suggestion
   - during the challenge period and challenge timeout period:
```

     - if the claim is challenged, and no more than `challengeTimoutPeriod` seconds have passed since the challenge, the arbitrator can call `approveClaim`. The arbitrator can change the bountyPercentage if given the permission

>    - during the challenge timeout period:

Reviewer: ```suggestion
   - during the challenge timeout period:
```

>      - if the claim was not challenged, anyone can call `approveClaim` and approve the claim.  In that case, the bountyPercentage remains that as chosen by the committee

Reviewer: ```suggestion
    - if the challenge period had passed and the claim was not challenged (but is not yet expired), anyone can call `approveClaim` and approve the claim. In that case, the bountyPercentage remains that as chosen by the committee
```

1. **DISMISSAL** `dismissClaim(_claimId)`
    - during the challenge period and challenge timeout period:
      - if the claim was challenged, the arbitrator can dismiss the claim
    - after challengePeriod + challengeTimeoutPeriod, the claim is _expired_  and:
      - anyone can dismiss the claim

During the time from submitting a claim to its resolution (i.e. approval or dismissal), the vault will be locked for withdrawals, and no new claims can be submitted

## Relevant settings

- `challengePeriod`: length of challlenge period, in seconds
- `challengeTimeoutPeriod`: length of time a challenged claim can be approved

> - `arbitratorCanChangeBounty`: determines whether the arbitrator change the bounty amount when approving a claim

Reviewer: ```suggestion
- `arbitratorCanChangeBounty`: determines whether the arbitrator can change the bounty percentage when approving a claim
```
